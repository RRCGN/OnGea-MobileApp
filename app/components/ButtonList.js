/**
 * Lists Buttons in a row
 */

import React from 'react'
import { View } from 'react-native'

const ButtonList = ({ children }) => {
  return (
    <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
      {children}
    </View>
  )
}

export default ButtonList
