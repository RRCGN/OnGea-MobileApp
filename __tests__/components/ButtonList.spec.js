import { View } from 'react-native'
import React from 'react'
import ButtonList from '../../app/components/ButtonList'

import renderer from 'react-test-renderer'

describe('<ButtonList />', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <ButtonList>
        <View />
        <View />
      </ButtonList>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with custom justifyContent', () => {
    const tree = renderer.create(
      <ButtonList justifyContent="flex-start">
        <View />
        <View />
      </ButtonList>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
