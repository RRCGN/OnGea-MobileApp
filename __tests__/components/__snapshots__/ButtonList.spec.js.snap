exports[`<ButtonList /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }>
  <View
    style={
      Object {
        "marginRight": 8,
      }
    }>
    <View />
  </View>
  <View
    style={
      Object {
        "marginRight": 8,
      }
    }>
    <View />
  </View>
</View>
`;

exports[`<ButtonList /> renders correctly with custom justifyContent 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }>
  <View
    style={
      Object {
        "marginRight": 8,
      }
    }>
    <View />
  </View>
  <View
    style={
      Object {
        "marginRight": 8,
      }
    }>
    <View />
  </View>
</View>
`;
